scanapp/serializers.py

```diff
@@ -0,0 +1,145 @@
+from scanapp.models import ScanError
+
+from rest_framework import serializers
+
+
+class ScanSerializer(serializers.ModelSerializer):
+    """
+    ModelSerializer for `Scan` model with all fields
+    """
+    class Meta:
+        model = Scan
+        fields = '__all__'
+
+
+class ScannedFileSerializer(serializers.ModelSerializer):
+    """
+    ModelSerializer for `ScannedFile` with all fields excluding `id` and `scan`
+    """
+    class Meta:
+        model = ScannedFile
+        exclude = ('id', 'scan')
+
+
+class LicenseSerializer(serializers.ModelSerializer):
+    """
+    ModelSerializer for `License` with all fields excluding `id` and `scanned_file`
+    """
+    class Meta:
+        model = License
+        exclude = ('id', 'scanned_file')
+
+
+class CopyrightSerializer(serializers.ModelSerializer):
+    """
+    ModelSerializer for `Copyright` with all fields excluding `id` and `scanned_file`
+    """
+    class Meta:
+        model = Copyright
+        exclude = ('id', 'scanned_file')
+
+
+class CopyrightHolderSerializer(serializers.ModelSerializer):
+    """
+    ModelSerializer for `CopyrightHolders` with all fields excluding `id` and `copyright`
+    """
+    class Meta:
+        model = CopyrightHolder
+        exclude = ('id', 'copyright')
+
+
+class CopyrightStatementSerializer(serializers.ModelSerializer):
+    """
+    ModelSerializer for `CopyrightHolders` with all fields excluding `id` and `copyright`
+    """
+    class Meta:
+        model = CopyrightStatement
+        exclude = ('id', 'copyright')
+
+
+class CopyrightAuthorSerializer(serializers.ModelSerializer):
+    """
+    ModelSerializer for `CopyrightAuthor` with all fields excluding `id` and `copyright`
+    """
+    class Meta:
+        model = CopyrightAuthor
+        exclude = ('id', 'copyright')
+
+
+class PackageSerializer(serializers.ModelSerializer):
+    """
+    ModelSerializer for `Package` with all fields excluding `id` and `scanned_file`
+    """
+    class Meta:
+        model = Package
+        exclude = ('id', 'scanned_file')
+
+
+class ScanErrorSerializer(serializers.ModelSerializer):
+    """
+    ModelSerializer for `Package` with all fields excluding `id` and `scanned_file`
+    """
+    class Meta:
+        model = ScanError
+        exclude = ('id', 'scanned_file')
+
+
+class AllModelSerializer(serializers.Serializer):
+    """
+    Another good serializer to handle all the serialization activities
+    """
+    scan = ScanSerializer()
+    scanned_file = ScannedFileSerializer(many=True)
+    license = LicenseSerializer(many=True)
+    copyright = CopyrightSerializer(many=True)
+    copyright_holder = CopyrightHolderSerializer(many=True)
+    copyright_statement = CopyrightStatementSerializer(many=True)
+    copyright_author = CopyrightAuthorSerializer(many=True)
+    package = PackageSerializer(many=True)
+    scan_error = ScanErrorSerializer(many=True)
+
+
+class AllModelSerializerHelper(object):
+    def __init__(self, scan):
```

Comment:
I have added the doc string to this particular class. BTW, it takes the scan_id from the URL and creates nested objects of results into various models.